### PR TITLE
#5: Basic UI test

### LINKED ISSUE #5: Basic UI Test
## Situation

IntelliJ plugins can be tested on two different levels: model/controller or directly on the UI.

UI tests are required to see if menu entries are there for example.#4 

We need a basic UI tests to get the initial setup right.

## Acceptance Criteria

1. UI tests proves that the menu entry for opening the OFT user guide exists.

### PR DESCRIPTION
**Pull request by redcatbear**
This PR adds a basic UI test. It cannot be fully automated yet, because there is no syncrhonization between the Gradle Task that startes IntelliJ with the Remote Robot server and the actual test task.

Trying to wait for the server to become ready inside the test code caused a race condition in Gradle. --> #6

Closes #5.

### CODE REVIEW

**Review by kaklakariada — CHANGES_REQUESTED** [review]
File `.idea/sonarlint/issuestore/0/3/0398ccd0f49298b10a3d76a47800d2ebecd49859` checked in by accident?

File: src/test/java/org/itallcode/openfasttrace/intelijplugin/uitest/PluginUiTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.nio.file.Path;
+import java.time.Duration;
+
+import static org.itallcode.openfasttrace.intelijplugin.remoterobot.RemoteRobotProperties.*;
+
+class PluginUiTest {
+    final static Duration WITH_PATIENCE = Duration.ofSeconds(10);
```
Comment (kaklakariada) [inline src/test/java/org/itallcode/openfasttrace/intelijplugin/uitest/PluginUiTest.java:21]: 😆 
10s is really long :-(

File: src/test/java/org/itallcode/openfasttrace/intelijplugin/uitest/pages/IdeFrameFixture.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ */
+@DefaultXpath(by = "IdeFrameImpl type", xpath = "//div[@class='IdeFrameImpl']")
+@FixtureName(name = "IDE Frame")
+public class IdeFrameFixture extends CommonContainerFixture {
+    public IdeFrameFixture(@NotNull RemoteRobot remoteRobot, @NotNull RemoteComponent remoteComponent) {
+        super(remoteRobot, remoteComponent);
+    }
+
+    public JMenuBarFixture menuBar() {
+        return find(JMenuBarFixture.class, byXpath("//div[@class='LinuxIdeMenuBar']"), Duration.ofSeconds(20));
```
Comment (kaklakariada) [inline src/test/java/org/itallcode/openfasttrace/intelijplugin/uitest/pages/IdeFrameFixture.java:23]: So this only works on Linux ;)

File: src/test/java/org/itallcode/openfasttrace/intelijplugin/uitest/pages/NewProjectDialogFixture.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package org.itallcode.openfasttrace.intelijplugin.uitest.pages;
+
+import com.intellij.remoterobot.RemoteRobot;
+import com.intellij.remoterobot.data.RemoteComponent;
+import com.intellij.remoterobot.fixtures.*;
+import org.jetbrains.annotations.NotNull;
+
+import static com.intellij.remoterobot.search.locators.Locators.byXpath;
+
+@DefaultXpath(by = "MyDialog type", xpath = "//*[contains(@title.key, 'title.new.project')]")
```
Comment (kaklakariada) [inline src/test/java/org/itallcode/openfasttrace/intelijplugin/uitest/pages/NewProjectDialogFixture.java:10]:
```suggestion
@DefaultXpath(by = "NewProjectDialog type", xpath = "//*[contains(@title.key, 'title.new.project')]")
```
❓

**Review by kaklakariada — CHANGES_REQUESTED** [review]

**Comment by kaklakariada** [comment]
Please also document how to start the tests manually

File: src/test/java/org/itallcode/openfasttrace/intelijplugin/uitest/pages/IdeFrameFixture.java (hunk 2 above)
Comment (redcatbear) [reply src/test/java/org/itallcode/openfasttrace/intelijplugin/uitest/pages/IdeFrameFixture.java:23]: At the moment: yes.

**Review by kaklakariada — DISMISSED** [review]

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**